security-analysis/security-analysis-default/src/main/java/com/powsybl/security/impl/DefaultSecurityAnalysis.java

```diff
@@ -212,6 +220,13 @@ private void setContingencyOkAndCheckViolations(String postContVariantId, Securi
         SecurityAnalysisResultBuilder.PostContingencyResultBuilder builder = resultBuilder.contingency(contingency).setComputationOk(lfResult.isOk());
         if (lfResult.isOk()) {
             violationDetector.checkAll(contingency, network, builder::addViolation);
+            monitors.forEach(stateMonitor -> {
```

Comment:
Here we will have a performance issue because of the linear search (if we have 10000 contingencies and 10000 "monitors", it will take forever).

In order to solve this, we must probably start by translating, at the start of the computaiton, the input list into an indexed object, so that here we just have to call a more efficient `StateMonitor contingencyMonitor = index.getMonitor(contingencyId);`